### PR TITLE
Replace all instances of fmt.Errorf(%v) with fmt.Errorf(%w)

### PR DESCRIPTION
**Pull request by delvh**
Found using
`find . -type f -name '*.go' -print -exec vim {} -c ':%s/fmt\.Errorf(\(.*\)%v\(.*\)err/fmt.Errorf(\1%w\2err/g' -c ':wq' \;`

### CODE REVIEW

File: models/git/branches.go [hunk 1]
```diff
@@ -487,7 +487,7 @@ func RemoveDeletedBranchByID(repoID, id int64) (err error) {
 	if affected, err := db.GetEngine(db.DefaultContext).Delete(deletedBranch); err != nil {
 		return err
 	} else if affected != 1 {
-		return fmt.Errorf("remove deleted branch ID(%v) failed", id)
+		return fmt.Errorf("remove deleted branch ID(%w) failed", id)
```
Comment (wxiaoguang) [inline models/git/branches.go]: But it's wrong here ........ and maybe many other places.

Comment (delvh) [reply models/git/branches.go]: Oh yeah, you are absolutely right.
Wait a second, I'll try a more restrictive search.

File: modules/log/log.go [hunk 2]
```diff
@@ -219,9 +219,9 @@ func ReleaseReopen() error {
 		logger := value.(*MultiChannelledLogger)
 		if err := logger.ReleaseReopen(); err != nil {
 			if accumulatedErr == nil {
-				accumulatedErr = fmt.Errorf("Error reopening %s: %v", key.(string), err)
+				accumulatedErr = fmt.Errorf("Error reopening %s: %w", key.(string), err)
 			} else {
-				accumulatedErr = fmt.Errorf("Error reopening %s: %v & %v", key.(string), err, accumulatedErr)
+				accumulatedErr = fmt.Errorf("Error reopening %s: %v & %w", key.(string), err, accumulatedErr)
```
Comment (wxiaoguang) [inline modules/log/log.go:224]: And here, it has 2 errors 😂

Maybe it's fine to only wrap the last one.

**Comment by silverwind** [comment]
IIRC, golangci-lint has a rule to detect this. We should enable it.

**Comment by delvh** [comment]
Agreed. But please only once this PR is merged.
In the next few minutes, a force-push will come that takes care of handling multiple errors and no error at all.
(force-push as I won't revert the commit simply to run the other search, and instead)
Surprisingly, I seem to have a found a simple regex that also takes care of the edge cases.

File: modules/log/log.go (hunk 2 above)
Comment (delvh) [reply modules/log/log.go:224]: Yeah, I just looked at the logic here and the other same occurrence in `./event.go`, and surprisingly, in both times my regex has produced the correct result.

**Comment by zeripath** [comment]
Oh gods there are some delightfully unenlightening error messages there.

@wxiaoguang I think you're right we're just going to have to go down the route of adding the stack trace to these.

**Comment by delvh** [comment]
What exactly do you mean?

**Comment by silverwind** [comment]
Found the linter: https://github.com/polyfloyd/go-errorlint

I think you can just add `errorlint` to `.golangci.yml` after all errors are fixed and the linter will ensure no new issues arise.

**Comment by zeripath** [comment]
@delvh not really related to this PR but look at some of the errors reported, e.g. in notification.go:loadRepo(...):

```go
	if n.Repository == nil {
		n.Repository, err = repo_model.GetRepositoryByIDCtx(ctx, n.RepoID)
		if err != nil {
			return fmt.Errorf("getRepositoryByID [%d]: %w", n.RepoID, err) // <- BUT WHY WAS I TRYING TO GET THIS?
		}
	}
```

If we think about this message a bit more - The error that GetRepositoryByIDCtx should return should already be `GetRepositoryByIDCtx[%d]: %w` and we shouldn't have to be adding that here. The error returned by this function should be adding the context that it's from loadRepo().

We're just adding the wrong context and doing so at every level.

I should add that at least this error is giving the context of the RepoID that is failing to be got from the DB. Many don't even provide that kind of information, e.g.:

```go
// DeleteGPGKey deletes GPG key information in database.
func DeleteGPGKey(doer *user_model.User, id int64) (err error) {
	key, err := GetGPGKeyByID(id)
	if err != nil {
		if IsErrGPGKeyNotExist(err) {
			return nil
		}
		return fmt.Errorf("GetPublicKeyByID: %w", err) // WHICH BLOODY KEY?
	}
```

**Comment by 6543** [comment]
I did test `errorlint` enabled and it catched a lot of good issues ... - but to many to enable it within this pull

**Review by 6543 — APPROVED** [review]

File: modules/gitgraph/graph_test.go [hunk 3]
```diff
@@ -270,446 +270,446 @@ func TestCommitStringParsing(t *testing.T) {
...
-| * | | | 
-| * | | | 
-| * | | | 
-| | * | | 
-| | | * | 
-| | |/ /  
-| |/| |   
-* | | | 
-|\| | | 
-| * | |
```
Comment (6543) [inline modules/gitgraph/graph_test.go]: unrelated

**Review by 6543 — CHANGES_REQUESTED** [review]
as comment

File: modules/gitgraph/graph_test.go (hunk 3 above)
Comment (delvh) [reply modules/gitgraph/graph_test.go]: Interesting. I have configured my Vim to strip out trailing spaces?
Why is there trailing space in the first place?
But I can remove it of course.

File: models/migrations/v210.go [hunk 4]
```diff
@@ -144,7 +144,7 @@ func remigrateU2FCredentials(x *xorm.Engine) error {
 				if !has {
 					has, err := sess.Where("`lower_name`=?", remigrated.LowerName).And("`user_id`=?", remigrated.UserID).Exist(new(webauthnCredential))
 					if err != nil {
-						return fmt.Errorf("unable to check webauthn_credential[lower_name: %s, user_id:%v]. Error: %w", remigrated.LowerName, remigrated.UserID, err)
+						return fmt.Errorf("unable to check webauthn_credential[lower_name: %s, user_id:%w]. Error: %w", remigrated.LowerName, remigrated.UserID, err)
```
Comment (zeripath) [inline models/migrations/v210.go]:
```suggestion
						return fmt.Errorf("unable to check webauthn_credential[lower_name: %s, user_id: %s]. Error: %w", remigrated.LowerName, remigrated.UserID, err)
```

**Comment by zeripath** [comment]
```
 ❯ git diff main  | grep '%w.*%w'                                                             [19:44:36]
+						return fmt.Errorf("unable to check webauthn_credential[lower_name: %s, user_id:%w]. Error: %w", remigrated.LowerName, remigrated.UserID, err)
+			return fmt.Errorf("failed to rollback directory change during failed username change from: %s to: %s. DB Error: %w. Filesystem Error: %w", oldUserName, newUserName, err, err2)
+				err = fmt.Errorf("%w %w", err2, err)
+			return fmt.Errorf("%w other names: %w", err, l.MultiChannelledLog.GetEventLoggerNames())
```

**Review by zeripath — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Generally L-G-T-M.

CI failures are related, golangci-lint does very good job and catches mismatched formatting arguments.

File: services/repository/files/upload.go [hunk 5]
```diff
@@ -42,7 +42,7 @@ func cleanUpAfterFailure(infos *[]uploadInfo, t *TemporaryUploadRepository, orig
 		}
 		if !info.lfsMetaObject.Existing {
 			if _, err := git_model.RemoveLFSMetaObjectByOid(t.repo.ID, info.lfsMetaObject.Oid); err != nil {
-				original = fmt.Errorf("%v, %v", original, err)
+				original = fmt.Errorf("%v, %w", original, err) // TODO: Do we wrap the parameter (original) here, or the new error?
```
Comment (6543) [inline services/repository/files/upload.go]: good question

**Review by 6543 — APPROVED** [review]

File: services/repository/files/upload.go (hunk 5 above)
Comment (zeripath) [reply services/repository/files/upload.go]: Wrap the original:

```suggestion
				original = fmt.Errorf("%w, %v", original, err) // We wrap the original error - as this is the underlying error that required the fallback
```

**Comment by zeripath** [comment]
There was an incorrect whitespace change in blame_test.go.

**Comment by lunny** [comment]
CI failure is related
